### PR TITLE
ISPN-14724 Create a simple DSL to build Protocol Buffers schema

### PR DESCRIPTION
**Pull request by fax4ever**
https://issues.redhat.com/browse/ISPN-14724

### CODE REVIEW

**Comment by tristantarrant** [comment]
I would put this in the infinispan-api module so it can be used in both embedded and remote deployments

**Comment by fax4ever** [comment]
Moved to the api module and added a real test.

**Review by tristantarrant — CHANGES_REQUESTED** [review]

File: core/src/test/java/org/infinispan/marshall/protostream/builder/ProtoBufBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                  .keyword()
+                     .normalizer("lowercase")
+               .optional("description", 3, "string")
+                  .text()
+                     .analyzer("english")
+               .required("author", 1, "org.infinispan.Author")
+                  .embedded()
+      .build();
+
+      assertThat(proto).isNotBlank();
```
[the remark this answers is not in the document]
Comment (fax4ever) [reply core/src/test/java/org/infinispan/marshall/protostream/builder/ProtoBufBuilderTest.java]: This file was supposed to be deleted, but I forgot to do that. I removed it.
The process is tested in the HotRod client module using our validation system and I just added the file comparison.

**Comment by fax4ever** [comment]
Thanks @tristantarrant, I think I fixed it.

**Comment by tristantarrant** [comment]
Merged, thanks

**Comment by fax4ever** [comment]
Thank you

**Pull request merged**